vm/vm.go

```diff
@@ -297,9 +297,16 @@ func (vm *VM) Initialize(
 		return nil, nil, nil, false, fmt.Errorf("failed to apply options : %w", err)
 	}
 
-	vm.chainTimeValidityWindow = validitywindow.NewTimeValidityWindow(vm.snowCtx.Log, vm.tracer, vm, func(timestamp int64) int64 {
-		return vm.ruleFactory.GetRules(timestamp).GetValidityWindow()
-	})
+	executionBlockParser := chain.NewBlockParser(vm.Tracer(), vm.txParser)
+
+	if err := vm.initChainStore(executionBlockParser); err != nil {
+		return nil, nil, nil, false, err
+	}
+
+	if err := vm.initValidityWindow(ctx); err != nil {
+		return nil, nil, nil, false, err
+	}
```

Comment:
We cannot guarantee within `vm.Initialize` that the validity window is fully populated.

If we move this into `startNormalOp`, then we CAN guarantee that some code we define ie. `initValidityWindow` executes and indicates that we successfully filled the validity window before entering normal operation.